server/src/test/integration/org/apache/cassandra/sidecar/routes/NodeDecommissionIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+        final String[] jobId = new String[1];
+        String testRoute = "/api/v1/cassandra/node/decommission";
+        testWithClient(client -> client.put(server.actualPort(), "127.0.0.1", testRoute)
+                                       .send(context.succeeding(response -> {
+                                           logger.info("Response Status:" + response.statusCode());
+                                           NodeDecommissionResponse decommissionResponse = response.bodyAsJson(NodeDecommissionResponse.class);
+                                           assertThat(decommissionResponse.status()).isEqualTo(RUNNING);
+                                           jobId[0] = String.valueOf(decommissionResponse.jobId());
+                                       })));
+        Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+        pollStatusForState(context, jobId[0], SUCCEEDED, null);
+        context.awaitCompletion(2, TimeUnit.MINUTES);
+    }
+
+    @CassandraIntegrationTest(nodesPerDc = 5, network = true, buildCluster = false)
+    void decommissionNodeWithFailure(VertxTestContext context,
+                                     ConfigurableCassandraTestContext cassandraTestContext) throws InterruptedException
+    {
+        BBHelperDecommissionNode.reset();
+        cassandraTestContext.configureAndStartCluster(builder -> {
+            builder.withInstanceInitializer(BBHelperDecommissionNode::install);
+        });
+
+        final String[] jobId = new String[1];
+        String testRoute = "/api/v1/cassandra/node/decommission";
+        testWithClient(client -> client.put(server.actualPort(), "127.0.0.1", testRoute)
+                                       .send(context.succeeding(response -> {
+                                           logger.info("Response Status:" + response.statusCode());
+                                           NodeDecommissionResponse decommissionResponse = response.bodyAsJson(NodeDecommissionResponse.class);
+                                           assertThat(decommissionResponse.status()).isEqualTo(RUNNING);
+                                           jobId[0] = String.valueOf(decommissionResponse.jobId());
+                                       })));
+        Uninterruptibles.sleepUninterruptibly(10, TimeUnit.SECONDS);
+        pollStatusForState(context, jobId[0], FAILED, DECOMMISSION_FAILED_MESSAGE);
+        context.awaitCompletion(2, TimeUnit.MINUTES);
+    }
+
+    private void pollStatusForState(VertxTestContext context,
+                                    String uuid,
+                                    OperationalJobStatus expectedStatus,
+                                    String expectedReason)
+    {
+
+        int attempts = 10;
+        String status = "/api/v1/cassandra/operational-jobs/" + uuid;
+        AtomicBoolean stateReached = new AtomicBoolean(false);
+        logger.info("Job Stats Attempt:" + attempts);
+        final int[] counter = {0};
```

Comment:
how about just use `AtomicInteger`? We do not need the atomicity property here, but its API is way more expressive than `counter[0]++`.